nextflow.config

```
@@ -1,154 +1,287 @@
 /*
- * -------------------------------------------------
- *  nf-core/circrna Nextflow config file
- * -------------------------------------------------
- * Default config options for all environments.
- */
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    nf-core/circrna Nextflow config file
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    Default config options for all compute environments
+----------------------------------------------------------------------------------------
+*/
 
 // Global default params, used in configs
+// TODO: tidy up after workflow is running smoothly
 params {
 
-  // Workflow flags
-  // TODO nf-core: Specify your pipeline's command line flags
-  genome = false
-  input = "data/*{1,2}.fastq.gz"
-  single_end = false
-  outdir = './results'
-  publish_dir_mode = 'copy'
-
-  // Boilerplate options
-  name = false
-  multiqc_config = false
-  email = false
-  email_on_fail = false
-  max_multiqc_email_size = 25.MB
-  plaintext_email = false
-  monochrome_logs = false
-  help = false
-  igenomes_base = 's3://ngi-igenomes/igenomes/'
-  tracedir = "${params.outdir}/pipeline_info"
-  igenomes_ignore = false
-  custom_config_version = 'master'
-  custom_config_base = "https://raw.githubusercontent.com/nf-core/configs/${params.custom_config_version}"
-  hostnames = false
-  config_profile_description = false
-  config_profile_contact = false
-  config_profile_url = false
-
-  // Defaults only, expecting to be overwritten
-  max_memory = 128.GB
-  max_cpus = 16
-  max_time = 240.h
+    // Input options
+    input                      = null
+    outdir                     = './results'
+    phenotype                  = null
 
-}
+    // workflow options
+    module                     = 'circrna_discovery'
+    tool                       = 'circexplorer2'
+    bsj_reads                  = 0
+    tool_filter                = 0
+    duplicates_fun             = 'mean'
+    exon_boundary              = 200
+    save_intermediates         = true
```

This is a bit awkard behaviour, as a user would have to specify `--save_intremediates false` if they don't want it, despite it being a boolean flag.

I would actually argue _not_ saving intermediates is preferable as it saves on HDD space. If you want it saved by default I would change the param name to `--discard_intermediates = false`

```suggestion
    save_intermediates         = false
```

---

fair, I think I had it on by default when converting to DSL2. false is the default 👍🏼